pkg/decision/composite_experiment_service.go

```diff
@@ -27,23 +27,57 @@ import (
 
 var ceLogger = logging.GetLogger("CompositeExperimentService")
 
+type CESOptionFunc func(*CompositeExperimentService)
+
+func WithUserProfileService(userProfileService UserProfileService) CESOptionFunc {
+	return func(f *CompositeExperimentService) {
+		f.userProfileService = userProfileService
+	}
+}
+
+func WithOverrideStore(overrideStore ExperimentOverrideStore) CESOptionFunc {
+	return func(f *CompositeExperimentService) {
+		f.overrideStore = overrideStore
+	}
+}
+
 // CompositeExperimentService bridges together the various experiment decision services that ship by default with the SDK
 type CompositeExperimentService struct {
 	experimentServices []ExperimentService
+	overrideStore      ExperimentOverrideStore
+	userProfileService UserProfileService
```

Comment:
Nit: it's a little weird that these fields are just use to move values from the option functions to a place where `NewCompositeExperimentService` can use them. Not sure there is a better alternative though.

Reply:
Yeah it was weird to me but at least they are private